examples/02_decoding/plot_haxby_understand_decoder.py

```diff
@@ -0,0 +1,324 @@
+# of selected trials, each corresponding to one of the 8 labels we selected
+# above.
+#
+# :class:`nilearn.decoding.Decoder` can convert this 4D image to a 2D numpy
+# array where each row corresponds to a trial and each column corresponds to a
+# voxel. In addition, it can also do several other things like masking,
+# smoothing, standardizing the data etc. depending on your requirements.
+#
+# Under the hood, :class:`nilearn.decoding.Decoder` uses
+# :class:`nilearn.maskers.NiftiMasker` to do all these operations. So here we
+# will demonstrate this by directly using the
+# :class:`nilearn.maskers.NiftiMasker`. Specifically, we will use it to:
+#
+# 1. only keep the data from the Ventral Temporal cortex by providing the
+# mask image (in :class:`nilearn.decoding.Decoder` this is done by
+# providing the mask image in the ``mask`` parameter).
+#
+# 2. standardize the data by z-scoring it such that the data is scaled to
+# have zero mean and unit variance across trials (in
+# :class:`nilearn.decoding.Decoder` this is done by setting the ``standardize``
+# parameter to ``"zscore_sample"``).
+
+from nilearn.maskers import NiftiMasker
+
+masker = NiftiMasker(mask_img=mask_vt, standardize="zscore_sample")
+
+# %%
+# Convert the multi-class labels to binary labels
+# -----------------------------------------------
+#
+# The :class:`nilearn.decoding.Decoder` converts multi-class classification
+# problem to N one-vs-others binary classification problems by default (where N
+# is the number of unique labels)
+#
+# The advantage of this approach is its interpretability. Once we are done with
+# training and cross-validating, we will have N AU-ROC scores, one for each
+# label. This will give us an insight into which labels (and the corresponding
+# cognitive domains) are easier to predict and are hence well differentiated
+# relative to the others in the brain.
+#
+# In addition, we will also have access to the classifier coefficients for each
+# label. These can be further used to understand the importance of each voxel
+# for each corresponding cognitive domain.
+#
+# In this example we have N = 8 unique labels and we will use Sklearn's
+# :class:`~sklearn.preprocessing.LabelBinarizer` to do this conversion.
+
+from matplotlib import pyplot as plt
+from sklearn.preprocessing import LabelBinarizer
+
+label_binarizer = LabelBinarizer(pos_label=1, neg_label=-1)
+y_binary = label_binarizer.fit_transform(y)
+
+# %%
+# Let's plot the labels to understand the conversion
+from matplotlib.colors import ListedColormap
+from sklearn.preprocessing import LabelEncoder
+
+# create a copy of y_binary and manipulate it just for plotting
+y_binary_ = y_binary.copy()
+for col in range(y_binary_.shape[1]):
+    y_binary_[np.where(y_binary_[:, col] == 1), col] = col
+
+fig, (ax_binary, ax_multi) = plt.subplots(
+    2, gridspec_kw={"height_ratios": [10, 1.5]}, figsize=(12, 2)
+)
+cmap = ListedColormap(["white"] + list(plt.cm.tab10.colors)[0:8])
```

Comment:
```suggestion
cmap = ListedColormap(["white"] + list(plt.cm.tab10.colors)[0:n_conditions])
``` where n_conditions has been set previously